src/game.cpp

```diff
@@ -1412,16 +1413,38 @@ bool game::do_turn()
 
     events.process();
     mission::process_all();
-    if( calendar::once_every( 1_days ) ) { // Midnight!
+
+    // Run a LUA callback once per day
+    if( calendar::once_every( 1_days ) ) // Midnight!
+    {
         overmap_buffer.process_mongroups();
-        lua_callback("on_day_passed");
+        // Run a LUA callback once per year
+        if( calendar::turn.day_of_year() == 0 ) {
+            // possible callback arguments: none (calendar is already exposed to lua)
+            lua_callback( "on_year_passed" );
+        }
+        // possible callback arguments: none (calendar is already exposed to lua)
+        lua_callback( "on_day_passed" );
     }
 
-    // Run a LUA callback once per minute
+    // Run a LUA callback once per hour
+    if( calendar::once_every( 1_hours ) ) {
+        // possible callback arguments: none (calendar is already exposed to lua)
+        lua_callback( "on_hour_passed" );
+    }
+ 
     if( calendar::once_every( 1_minutes ) ) {
+        // possible callback arguments: none (calendar is already exposed to lua)
         lua_callback("on_minute_passed");
     }
 
+    // Run a LUA callback once per turn
+    if( calendar::once_every( 1_turns ) ) {
```

Comment:
This check is a bit pointless as it will be `true` all the time. (`game::do_turn` is called once for each turn, it actually increments the turn counter.)